File: javascript/lesson 1/examples.js
function sum(nums) {
    let result = 0;
    for (let i = 0; i < nums.length; i++) {
        result += nums[i];
    }
    return result;
}

function countZeroSumNotOptimized(nums) {
    let result = 0;
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[i] + nums[j] == 0)
                result++;
        }
    }
    return result;
}

function sumRecursive(nums) {
    return sumRecursiveImpl(nums, 0);
}

function sumRecursiveImpl(nums, current) {
    if (current >= nums.length) return 0;

    return nums[current] + sumRecursiveImpl(nums, current + 1);
}

File: kotlin/lesson 1/Examples.kt
public class Examples {
    fun sum(nums: IntArray): Int {
        var result = 0
        for (i in nums.indices) {
            result += nums[i]
        }
        return result
    }

    fun countZeroSumNotOptimized(nums: IntArray): Int {
        var result = 0
        for (i in nums.indices) {
            for (j in i + 1 until nums.size) {
                if (nums[i] + nums[j] == 0)
                    result++
            }
        }
        return result
    }

    fun sumRecursive(nums: IntArray): Int {
        return sumRecursiveImpl(nums, 0)
    }

    fun sumRecursiveImpl(nums: IntArray, current: Int): Int {
        if (current >= nums.size) return 0

        return nums[current] + sumRecursiveImpl(nums, current + 1)
    }
}
